src/test/scala/com/feedzai/cosytest/CleanUp.scala

```diff
@@ -0,0 +1,74 @@
+package com.feedzai.cosytest
+
+import java.nio.file.Paths
+
+import org.scalatest.{BeforeAndAfterAll, TestSuite}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+/**
+ *   Trait used to clean up docker environment if tests leave system dirty.
+ *
+ *   Note: Shouldn't be used if tests run in parallel
```

Comment:
That was because when the test fails, I used a method to remove all containers and networks even if not started by test. I changed it and it now works as it should be.